### ISSUE TITLE
Can't turbo or tturbo enumerate blocks

### ISSUE THREAD

**Issue by OmegaLambda1998** [JuliaSIMD/LoopVectorization.jl#393]
Both of the following code blocks produce `ERROR: LoadError: TypeError: in typeassert, expected Symbol, got a value of type Expr`
```
@turbo for (i, n) in enumerate(1:100)
begin end
end
```

```
@tturbo for (i, n) in enumerate(1:100)
begin end
end
```
wrapping `enumerate` in `collect` does not resolve this issue

**Comment by chriselrod** [comment]
I'm focusing most of my free/SIMD time on the rewrite, which will not have this problem.
So I've largely stopped putting time into LoopVectorization.jl issues/bugs.
If I didn't, I'd never have the time to actually work on the replacement.

It'll likely still be a long time before it is ready to replace LoopVectorization (e.g. a year or so is my current estimate).

So all that said, this would be a good first issue if someone wants to try adding support for this themselves.

The basic approach would be to basically replace
```julia
for (i,n) in enumerate(x)

end
```
with
```julia
for i in 1:length(x)
    n = x[i+firstindex(x)-1]
end
```
If anyone is interested, I can provide a more detailed explanation of ideas for how to implement that in the code and answer any questions.

**Comment by chriselrod** [comment]
Fixed in https://github.com/JuliaSIMD/LoopVectorization.jl/pull/396

**Comment by wangl-cc** [comment]
Hi, the PR #396 doesn't seems to fix this issue. @chriselrod

**Comment by chriselrod** [comment]
> Hi, the PR #396 doesn't seems to fix this issue. @chriselrod

I closed the wrong issue by mistake.

**Comment by chriselrod** [comment]
Closed by #397.
